xgi/linalg/matrix.py

```diff
@@ -425,19 +424,20 @@ def boundary_matrix(S, order=1, orientations=None, index=False):
         rowdict = {v: k for k, v in simplices_d_dict.items()}
         coldict = {v: k for k, v in simplices_u_dict.items()}
 
-    if orientations == None:
-        orientations = {
-            idd: 0 for idd in list(S.edges.filterby("order", 1, mode="geq"))
-        }
+    if orientations is None:
+        orientations = {idd: 0 for idd in S.edges.filterby("order", 1, mode="geq")}
 
     B = np.zeros((nd, nu))
     if not (nu == 0 or nd == 0):
         if order == 1:
             for u_simplex_id in simplices_u_ids:
-                u_simplex = S.edges.members(u_simplex_id)
+                u_simplex = list(S.edges.members(u_simplex_id))
+                u_simplex.sort(
+                    key=lambda e: (isinstance(e, str), e)
+                )  # Sort for reference orientation
```

Comment:
Given that this is a fairly surprising thing to do, please leave a comment in the code explaining why the tuple `(<bool>, e)` is necessary.

Reply:
Done